aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/ReadHandler.java

```diff
@@ -1,50 +1,112 @@
-import software.amazon.cloudformation.proxy.ProxyClient;
 import software.amazon.cloudformation.proxy.Logger;
 import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
 
 public class ReadHandler extends BaseHandlerStd {
-    private static final String OPERATION = "ReadParameter";
-
-    @Override
-    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-            final AmazonWebServicesClientProxy proxy,
-            final ResourceHandlerRequest<ResourceModel> request,
-            final CallbackContext callbackContext,
-            final ProxyClient<SsmClient> proxyClient,
-            final Logger logger) {
-        return proxy.initiate("aws-ssm-parameter::resource-read", proxyClient, request.getDesiredResourceState(), callbackContext)
-                .translateToServiceRequest(Translator::getParametersRequest)
-                .makeServiceCall(this::ReadResource)
-                .done((getParametersRequest, getParametersResponse, proxyInvocation, resourceModel, context) -> {
-                    if(getParametersResponse.parameters().size() == 0) {
-                        throw new CfnNotFoundException(ResourceModel.TYPE_NAME, request.getDesiredResourceState().getName());
-                    }
-                    final Parameter parameter = getParametersResponse.parameters().stream().findFirst().get();
-
-                    return ProgressEvent.defaultSuccessHandler(ResourceModel.builder()
-                            .name(parameter.name())
-                            .type(parameter.typeAsString())
-                            .value(parameter.value()).build());
-                });
-    }
-
-    private GetParametersResponse ReadResource(final GetParametersRequest getParametersRequest,
-                                               final ProxyClient<SsmClient> proxyClient) {
-        try{
-            return proxyClient.injectCredentialsAndInvokeV2(getParametersRequest, proxyClient.client()::getParameters);
-        } catch (final InternalServerErrorException exception) {
-            throw new CfnServiceInternalErrorException(OPERATION, exception);
-        }
-    }
+
+	public ReadHandler() {
+		super();
+	}
+
+	@VisibleForTesting
+	protected ReadHandler(SsmClient ssmClient) {
+		super(ssmClient);
+	}
+
+	@Override
+	protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+		final AmazonWebServicesClientProxy proxy,
+		final ResourceHandlerRequest<ResourceModel> request,
+		final CallbackContext callbackContext,
+		final ProxyClient<SsmClient> proxyClient,
+		final Logger logger) {
+		final ResourceModel model = request.getDesiredResourceState();
+
+		logger.log("Invoking Read Handler");
+		logger.log("READ ResourceModel: " + model.toString());
+
+		return ProgressEvent.progress(model, callbackContext)
```

Comment:
Can you verify to see if Drift if working for values other than Type/Value? Looks like native is only checking for these 2 attributes.

Reply:
Drift does not work for Native. Yes Native only checks for 2 attributes. Current Uluru implementation supports all attributes.